packages/api-bindings/src/apollo/cache/utils/cursorBasedPagination.ts

```diff
@@ -27,34 +23,71 @@ export function cursorBasedPagination<T extends CursorBasedPagination>(
 
       // items that are not in the cache anymore (for .e.g deleted publication)
       const danglingItems = items.filter((item) => !canRead(item));
-      const readRes = {
+      const readRes: SafeReadonly<TResult> = {
         ...existing,
         items,
         pageInfo: {
           ...pageInfo,
+          beforeCount: existing.pageInfo.beforeCount ?? 0,
+          moreAfter: existing.pageInfo.moreAfter ?? existing.pageInfo.next !== null,
           // reduce total count by excluding dangling items so it won't cause a new page query
           // after item was removed from the cache (for .e.g deleted publication)
           totalCount:
             pageInfo.totalCount !== null ? pageInfo.totalCount - danglingItems.length : null,
         },
-      } as SafeReadonly<T>;
+      };
 
       return readRes;
     },
 
-    merge(existing: Readonly<T> | undefined, incoming: SafeReadonly<T>) {
-      if (!existing) {
-        return incoming;
+    merge(
+      existing: Readonly<TResult> | undefined,
+      incoming: SafeReadonly<TResult>,
+      { variables = {} }: FieldFunctionOptions,
+    ) {
+      if (!isCursor(variables.cursor) || !existing) {
+        return {
+          ...incoming,
+          pageInfo: {
+            ...incoming.pageInfo, // future-proofing in case we add more fields to pageInfo
+            beforeCount: 0, // assume there is no newer results than the provided
+            moreAfter: incoming.pageInfo.next !== null,
+          },
+        };
       }
 
       const existingItems = existing.items;
       const incomingItems = incoming.items;
 
-      return {
-        ...incoming,
-        items: existingItems.concat(incomingItems),
-        pageInfo: incoming.pageInfo,
-      } as SafeReadonly<T>;
+      if (variables.cursor === existing.pageInfo.prev) {
+        return {
+          ...incoming,
+          items: incomingItems.concat(existingItems),
+          pageInfo: {
+            ...incoming.pageInfo, // future-proofing in case we add more fields to pageInfo
+            beforeCount: incoming.pageInfo.beforeCount,
```

Comment:
In an previous iteration of this PR it was called `moreBefore`. Then I started to entertain the idea of building a functionality that would (via periodic polling) check for new items and propagate this information to the UI so that "Show X newer tweets" feature could be implemented. This is how `moreBefore` evolved into `beforeCount`. I then decided to descope the observing logic into [T-11159](https://avara.height.app/T-11159) cause I had more urgent tickets to focus on.

The reason i left most of the `beforeCount` (even if stays to `0` in this PR) is that there are some test cases that will become useful for that follow-up ticket. I just didn't want to delete them knowing I would pick up that ticket at some point next week.

Reply:
got it, makes sense

Reply:
love that we would expose `beforeCount`, makes the UI a lot better on the consumer side.